packages/model-viewer/src/test/features/scene-graph/material-spec.ts

```diff
@@ -117,4 +120,58 @@ suite('scene-graph/material', () => {
       expect(threeTexture.uuid).to.be.equal(newUUID);
     });
   });
+
+  suite('Material properties', () => {
+    let element: ModelViewerElement;
+
+    setup(async () => {
+      element = new ModelViewerElement();
+    });
+
+    teardown(() => {
+      document.body.removeChild(element);
+    });
+
+    const loadModel = async (path: string) => {
+      element.src = path;
+      document.body.insertBefore(element, document.body.firstChild);
+      await waitForEvent(element, 'load');
+    };
+
+    test('test alpha cutoff expect default', async () => {
+      await loadModel(HELMET_GLB_PATH);
+      // Default value hack see:
+      // https://github.com/google/model-viewer/blob/e3a000111980f5cf018c61dec8695463e0d843a0/packages/model-viewer/src/three-components/gltf-instance/ModelViewerGLTFInstance.ts#L210
+      expect(element.model!.materials[0].getAlphaCutoff())
+          .to.be.equal(ALPHA_TEST_DEFAULT);
+      element.model!.materials[0].setAlphaCutoff(0.25);
```

Comment:
Actually, I don't think this is right. It's fine to set the cutoff to zero; if you look at the code for that hack, this negative value is actually what's set when blend mode equals opaque.

Reply:
Sure that makes sense. But this edge case could be exposed given that we can now change the blend mode. Not that it makes sense in the first place, working around weird little worts is not ideal, I can just leave it out.

Reply:
Wait so this must mean three.js is applying masking even when blend-mode is opaque. I guess that makes sense given that masking is a shader snippet, seems like maybe it should be compiled out when updating the material though.

Reply:
Actually it's a little weirder. I believe the problem is that for three's opaque materials, if there's an alpha channel value, that value ends up in the output pixel color, which is weird on a transparent rendering. I think alpha cutoff = 0 is used by three to turn off the masking code in the shader. Instead, we use a negative value to turn it on and replace that shader chunk with our own: https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/three-components/shader-chunk/alphatest_fragment.glsl.ts

That is where the alpha gets set back to fully opaque for the output, in order to be glTF spec-compliant.

Reply:
Interesting, curious about the alpha channel value that ends up in the output color, is it just random garbage? There used to be a Unity bug like this.
This explains why the default blend-mode is always set to blending enabled.
So really negative alpha-mask is how we enable Opaque rendering.